### PR TITLE
[UX] remove all uses of deprecated `sky jobs`

### PR DESCRIPTION
**Pull request by cg505**
There are a few lingering uses of the `sky spot` CLI, which has been deprecated since v0.6.0. Replace them with `sky jobs`.

Tested (run the relevant ones):

- [ ] Code formatting: `bash format.sh`
- [ ] Any manual or new tests for this PR (please specify below)
- [ ] All smoke tests: `pytest tests/test_smoke.py` 
- [ ] Relevant individual smoke tests: `pytest tests/test_smoke.py::test_fill_in_the_name` 
- [x] Backward compatibility tests: `conda deactivate; bash -i tests/backward_compatibility_tests.sh`
  - starting from 7

### CODE REVIEW

**Review by romilbhardwaj — COMMENTED** [review]
Nice @cg505!

File: llm/vicuna-llama-2/README.md [hunk 1]
```diff
@@ -122,7 +122,7 @@ sky launch --no-use-spot ...
 
 [SkyPilot Managed Spot](https://skypilot.readthedocs.io/en/latest/examples/spot-jobs.html) is a library built on top of SkyPilot that helps users run jobs on spot instances without worrying about interruptions. That is the tool used by the LMSYS organization to train the first version of Vicuna (more details can be found in their [launch blog post](https://lmsys.org/blog/2023-03-30-vicuna/) and [example](https://github.com/skypilot-org/skypilot/tree/master/llm/vicuna)). With this, the training cost can be reduced from $1000 to **\$300**.
```
[the remark this answers is not in the document]
Comment (cg505) [reply llm/vicuna-llama-2/README.md:123]: Good catch!

**Comment by cg505** [comment]
Fixed a couple of other mentions of "spot jobs" in text: https://github.com/skypilot-org/skypilot/pull/4173/commits/2c71e571ef06cff98b7f684512afe0d1492bbc89

**Review by romilbhardwaj — APPROVED** [review]
Awesome, thanks @cg505!

**Pull request merged**